background-charm-service/src/space-manager.ts

```diff
@@ -242,16 +243,22 @@ export class SpaceManager {
   private recordSuccess(charmId: string, bg: Cell<BGCharmEntry>) {
     bg.update({
       lastRun: Date.now(),
+      status: "Success",
     });
     if (this.schedulableBgs.has(charmId)) {
       this.addPendingRun(charmId, bg, this.rerunIntervalMs / 1000);
     }
   }
 
-  private disableCharm(charmId: string, bg: Cell<BGCharmEntry>) {
+  private disableCharm(
+    charmId: string,
+    bg: Cell<BGCharmEntry>,
+    error?: string,
+  ) {
     bg.update({
       disabledAt: Date.now(),
       lastRun: Date.now(),
+      status: error ? error : "Disabled",
```

Comment:
nit: `error ?? "Disabled"`

Reply:
deno typescript doesn't like it:

```
Property 'error' does not exist on type '{ success: boolean; data?: any; charmId: string; } | { success: boolean; error: string; charmId: string; } | { success: boolean; error: string; }'.

  Property 'error' does not exist on type '{ success: boolean; data?: any; charmId: string; }'.

deno-ts(2339)
```

Reply:
`error` is a `string | undefined`, not sure what that type is

Reply:
![image](https://github.com/user-attachments/assets/4eed1a33-d53c-4058-86e0-7a990c320a50)

I think it might just be typescript being overly cautious?